src/ir/module-utils.h

```diff
@@ -390,29 +390,32 @@ template<typename T> struct CallGraphPropertyAnalysis {
   //
   // hasProperty() - Check if the property is present.
   // canHaveProperty() - Check if the property could be present.
-  // addProperty() - Adds the property. This receives a second parameter which
-  //                 is the function due to which we are adding the property.
+  // addProperty() - Adds the property.
+  // logVisit() - Log each visit of the propagation. This is called before
+  //                 we check if the function already has the property.
   void propagateBack(std::function<bool(const T&)> hasProperty,
                      std::function<bool(const T&)> canHaveProperty,
-                     std::function<void(T&, Function*)> addProperty,
+                     std::function<void(T&)> addProperty,
+                     std::function<void(const T&, Function*)> logVisit,
                      NonDirectCalls nonDirectCalls) {
     // The work queue contains items we just learned can change the state.
     UniqueDeferredQueue<Function*> work;
     for (auto& func : wasm.functions) {
       if (hasProperty(map[func.get()]) ||
           (nonDirectCalls == NonDirectCallsHaveProperty &&
            map[func.get()].hasNonDirectCall)) {
-        addProperty(map[func.get()], func.get());
```

Comment:
Ahh, I've worked out why it originally checked `verbose && !info.canChangeState` - otherwise it would log here, which we don't want.